src/internet-computer-programming-concepts/basic-memory-persistence/stable-variables.md

```diff
@@ -1,7 +1,10 @@
 # Stable Variables
-The *mutable state* of an actor is stored in the form of mutable variables that are declared with the `var` keyword. Mutable variables in actors always have `private` *visibility*. (although the `private` keyword is not necessary and is assumed by default).
+To persist the state of an actor when [upgrading](/internet-computer-programming-concepts/basic-memory-persistence/upgrades.html), we can declare [immutable](/common-programming-concepts/variables.html) and [mutable](/common-programming-concepts/mutability.html) variables to be `stable`. Stable variables must be of [*stable type*](/internet-computer-programming-concepts/basic-memory-persistence/stable-variables.html#stable-types).
 
-If we want to persist the state of our actor when [upgrading](/internet-computer-programming-concepts/basic-memory-persistence/upgrades.html), we could declare our mutable variables `stable`. A stable variable looks like this:
+## Mutating and Upgrading Stable Mutable Variables
+The [*mutable state of an actor*](/internet-computer-programming-concepts/actors.html#a-simple-actor) is stored in the form of mutable variables that are declared with the `var` keyword. Mutable variables in actors always have `private` *visibility*. (although the `private` keyword is not necessary and is assumed by default).
+
+If we want to persist the state of our actor when [upgrading](/internet-computer-programming-concepts/basic-memory-persistence/upgrades.html), we could declare our mutable variables `stable`, if it is of [stable type](/internet-computer-programming-concepts/basic-memory-persistence/stable-variables.html#stable-types). A stable variable looks like this:
 ```motoko
 {{#include _mo/stable-variables.mo:a}}
 ```
```

Comment:
```suggestion
## Modifying and Upgrading Stable Variables
The [*state of an actor*](/internet-computer-programming-concepts/actors.html#a-simple-actor) is stored in the form of variables that are declared with the `var` keyword or `let` keywords. Variable declarations in actors always have `private` *visibility*. (although the `private` keyword is not necessary and is assumed by default). In an actor, only `shared` functions can be declared `public`.  This means that a client of an actor can only communicate with the actor by calling its public functions, and never by directly modifying its private state variables, isolating actors from each other.

If we want to retain the state of our actor when [upgrading](/internet-computer-programming-concepts/basic-memory-persistence/upgrades.html), we need to declare all its state variables as `stable`. A variable can only be declared stable if  if it is of [stable type](/internet-computer-programming-concepts/basic-memory-persistence/stable-variables.html#stable-types). Many, but not all types, are stable. A stable variable declaration looks like this:
```motoko
{{#include _mo/stable-variables.mo:a}}
```
```

Stable variables can be both immutable and mutable, so I'd stress mutability less. An immutable stable variable might seem useless, but it may actually contain a reference to data that is itself mutable inside,
eg

```
actor counter{
   stable let counter = { var value = 1 };
   public shared func inc() : async () { counter.value := 1 };
}
```
here counter is let bound, so hard on the outside, yet chewy (mutable) on the inside. If you see what I mean.

Reply:
Got it!